examples/scale/frontend/frontend.go

```diff
@@ -60,73 +60,89 @@ func run(cfg config.View) {
 	}
 	fe := pb.NewFrontendServiceClient(conn)
 
-	w := logger.Writer()
-	defer w.Close()
-
 	ticketQPS := int(activeScenario.FrontendTicketCreatedQPS)
 	ticketTotal := activeScenario.FrontendTotalTicketsToCreate
 
-	for {
-		currentCreated := int(atomic.LoadUint32(&totalCreated))
-		if ticketTotal != -1 && currentCreated >= ticketTotal {
-			break
+	totalCreated := 0
+
+	for range time.Tick(time.Second) {
+		for i := 0; i < ticketQPS; i++ {
+			if ticketTotal == -1 || totalCreated < ticketTotal {
+				go runner(fe)
+			}
 		}
+	}
+}
 
-		// Each inner loop creates TicketCreatedQPS tickets
-		var ticketPerRoutine, ticketModRoutine int
-		start := time.Now()
+func runner(fe pb.FrontendServiceClient) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-		if ticketTotal == -1 || currentCreated+ticketQPS <= ticketTotal {
-			ticketPerRoutine = ticketQPS / numOfRoutineCreate
-			ticketModRoutine = ticketQPS % numOfRoutineCreate
-		} else {
-			ticketPerRoutine = (ticketTotal - currentCreated) / numOfRoutineCreate
-			ticketModRoutine = (ticketTotal - currentCreated) % numOfRoutineCreate
-		}
+	g := stateGauge{}
+	defer g.stop()
 
-		var wg sync.WaitGroup
-		for i := 0; i < numOfRoutineCreate; i++ {
-			wg.Add(1)
-			if i < ticketModRoutine {
-				go createPerCycle(&wg, fe, ticketPerRoutine+1, start)
-			} else {
-				go createPerCycle(&wg, fe, ticketPerRoutine, start)
-			}
-		}
+	g.start(mRunnersWaiting)
```

Comment:
Why is runners waiting an interesting metric? Isnt creating the only interesting one? Given that there is no failure condition between running / creating, wont every running get to creating? If there was only creating, could the usage of state gauge be avoided and we simply have mRunnersCreating(1) with a defer to mRunnersCreating(-1)?

Reply:
It's a little interesting, as it helps the conceptual model of what each runner is doing: waiting a little bit, then creating a ticket.  In the near future there will be other stages, waiting/polling for assignment, and deleting the ticket.

Reply:
IMO given there are no ways you fail, a measure of how long you wait before create may not tell us much (what would we infer between multiple runs - or about the system based on this metric?). I am ok it being in there for now - but as you revisit for other metrics, based on whether it is useful, please consider removing it. (sometimes having more clutter on the dashboard can be distracting :)